sdk/src/types/block/slot/index.rs

```diff
@@ -17,6 +20,14 @@ impl SlotIndex {
     pub fn new(index: u64) -> Self {
         Self::from(index)
     }
+
+    pub fn to_epoch_index(self, slots_per_epoch_exponent: u32) -> EpochIndex {
+        EpochIndex::new((self.0 >> slots_per_epoch_exponent) + 1)
```

Comment:
Why the +1 ?

Reply:
I don't know, but it's in the TIP

Reply:
Looks wrong to me

Reply:
i think epoch starts at 1 as 0 is reserved? something like that

Reply:
Which TIP is that please?

Reply:
https://github.com/iotaledger/tips-draft/blob/tip46/tips/TIP-0046/tip-0046.md#slot-index

Reply:
I've left a comment, I think it's wrong but we can edit later. I'll resolve this for now.